src/main/java/jenkins/plugins/openstack/compute/internal/Openstack.java

```diff
@@ -161,20 +164,75 @@ public Openstack(@Nonnull final OSClient client) {
         return nets;
     }
 
-    public @Nonnull Collection<Image> getSortedImages() {
-        List<? extends Image> images = clientProvider.get().images().listAll();
-        TreeSet<Image> set = new TreeSet<>(RESOURCE_COMPARATOR); // Eliminate duplicate names
-        set.addAll(images);
-        return set;
-    }
-
     private static final Comparator<BasicResource> RESOURCE_COMPARATOR = new Comparator<BasicResource>() {
         @Override
         public int compare(BasicResource o1, BasicResource o2) {
             return ObjectUtils.compare(o1.getName(), o2.getName());
         }
     };
 
+    /**
+     * Finds all {@link Image}s.
+     * 
+     * @return A Map of collections of images, indexed by name (or id if the
+     *         image has no name) in ascending order and, in the event of
+     *         name-collisions, the images for a given name are sorted by
+     *         creation date.
+     */
+    public @Nonnull Map<String, Collection<Image>> getImages() {
```

Comment:
Indirectly.
I wanted the method in Openstack.java that listed all the Images (getImages()) to have the same API as the one that lists all the VolumeSnapshots (getVolumeSnapshots()), and getVolumeSnapshotIdsFor(String) calls getVolumeSnapshots().
i.e. the new semantics are required for VolumeSnapshots but not Images, but Images was changed to keep it consistent.